addons/account/models/account_journal_dashboard.py

```diff
@@ -1,33 +1,37 @@
-import json
+import ast
+from babel.dates import format_datetime, format_date
+from collections import defaultdict
 from datetime import datetime, timedelta
+import json
+import random
 
-from babel.dates import format_datetime, format_date
 from odoo import models, api, _, fields
 from odoo.exceptions import UserError
 from odoo.osv import expression
 from odoo.release import version
 from odoo.tools import DEFAULT_SERVER_DATE_FORMAT as DF
 from odoo.tools.misc import formatLang, format_date as odoo_format_date, get_lang
-import random
 
-import ast
+
+def group_by_journal(vals_list):
```

Comment:
make it an inner function, it's used in a single place :)

Reply:
Not anymore in a single place